Update monasca-agent cookbook and barclamp

matrixik opened the pull request.

matrixik committed:
Add monasca-reconfigure and sudoers file

matrixik committed:
Fix monasca-setup bin path

matrixik committed:
Fix template variables

matrixik committed:
Keystone settings for agent

matrixik committed:
Keystone settings for agent

matrixik committed:
Keystone settings for agent

matrixik committed:
Add log

kornicameister commented on chef/cookbooks/monasca/recipes/agent.rb:
```diff
@@ -34,8 +34,109 @@
+  action :add_service
+end
+
+keystone_register "register monasca-agent endpoint" do
```
I think that needs to go api.rb, you install monasca-api you register an endpoint for it.

matrixik committed:
More readable config templates

matrixik committed:
Remove log

matrixik committed:
Remove per-node credentials todos

matrixik committed:
Fix json indentations

matrixik committed:
Rename agent dimensions variable name

matrixik committed:
Run monasca-reconfigure every time with chef client

matrixik committed:
Add firts GUI elements

matrixik committed:
Add GUI elements

matrixik commented:
I have problems with the GUI part:
```
======== Migrating barbican, ceilometer, ceph, cinder, cisco_ucs, crowbar, database, deployer, dns, glance, heat, horizon, hyperv, ipmi, keystone, logging, magnum, manila, network, neutron, nfs_client, nova, ntp, openstack, pacemaker, provisioner, rabbitmq, sahara, suse_manager_client, swift, tempest, trove, updater barclamps -- Tue Jan 31 09:32:42 2017 ========
WARNING: Nokogiri was built against LibXML version 2.9.1, but has dynamically loaded 2.9.4
active
rake aborted!
can not load translations from /opt/dell/crowbar_framework/config/locales/monasca/en.yml: #<Psych::SyntaxError: (/opt/dell/crowbar_framework/config/locales/monasca/en.yml): did not find expected key while parsing a block mapping at line 31 column 11>
/opt/dell/crowbar_framework/app/models/proposal.rb:13:in `<class:Proposal>'
/opt/dell/crowbar_framework/app/models/proposal.rb:1:in `<top (required)>'
/opt/dell/crowbar_framework/lib/schema_migration.rb:104:in `get_template_for_barclamp'
/opt/dell/crowbar_framework/lib/schema_migration.rb:34:in `run_for_bc'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:30:in `block (3 levels) in <top (required)>'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:29:in `each'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:29:in `block (2 levels) in <top (required)>'
Tasks: TOP => crowbar:schema_migrate
```

Edit:
Fixed it, I had to many `'` in one line

matrixik committed:
Update for new code arrangement for keystone

matrixik committed:
Fix yaml for translations

matrixik committed:
Fix monasca helper module name

matrixik committed:
Use root user for agent temporally

matrixik committed:
Update monasca agent GUI parts

matrixik committed:
WIP: Add custom checks for monasca-agent

matrixik committed:
Extract variables for templates

matrixik committed:
Fix testing filepath

matrixik committed:
Fix template for monasca agent checks

matrixik committed:
Fix template paths again

matrixik committed:
Address Hound warning

matrixik committed:
Add more monasca agent checkers

matrixik committed:
Fix some variables

matrixik committed:
Add log api port to data_bag

matrixik committed:
Build Kibana and log-api urls

matrixik committed:
Add more info to GUI

matrixik committed:
Add user_managed: false

matrixik committed:
Fix yaml indentation

matrixik committed:
Add some logs

matrixik committed:
Fix monasca-reconfigure

matrixik committed:
Fix ruby boolean name

matrixik committed:
Uncomment agent.yaml template code

matrixik committed:
Fix providing service variable to template

matrixik committed:
Fix insecure args for monasca-setup

matrixik committed:
Fix monasca-reconfigure

matrixik committed:
Configure more variables with monasca-reconfigure

matrixik committed:
Add more config to UI

matrixik committed:
Disable generating agent.yaml

It's regenerated every time chef client is run, it's running
monasca-reconfigure.

matrixik committed:
Change default overwrite config to false

matrixik committed:
Add monasca-agent to sudoers

artur-ba commented on chef/cookbooks/monasca/templates/default/conf.d/http_check.yaml.erb:
```diff
@@ -0,0 +1,14 @@
+    url: <%= @log_api_url %>/healthcheck
+    timeout: 1
+    dimensions:
+      component: monasca-log-api
```
Shouldn't be openstack-monasca-log-api?

matrixik replied:
Entire file removed

matrixik committed:
Run monasca-setup as monasca-agent user

matrixik committed:
Remove logs from monasca-reconfigure

kornicameister commented on chef/cookbooks/monasca/attributes/default.rb:
```diff
@@ -36,6 +40,12 @@
 default[:monasca][:api][:bind_host] = "*"
 
+# agent default service settings
+default[:monasca][:agent][:service_name] = monasca_agent_service_name
```
I am thinking if it'b better to go with **:metric_agent** just to match log agent ?

matrixik replied:
Replaced everywhere

kornicameister commented on chef/cookbooks/monasca/recipes/agent.rb:
```diff
@@ -14,38 +14,111 @@
+kibana_url = "http://#{monasca_public_host}:5601"
+log_api_url = "http://#{monasca_public_host}:#{node[:monasca][:log_api][:bind_port]}"
+
+# template "/etc/monasca/agent/agent.yaml" do
```
What's the purpose of these commented lines ?

matrixik replied:
Removed

kornicameister commented on chef/cookbooks/monasca/templates/default/conf.d/http_check.yaml.erb:
```diff
@@ -0,0 +1,14 @@
+    url: <%= @log_api_url %>/healthcheck
+    timeout: 1
+    dimensions:
+      component: monasca-log-api
```
+1

kornicameister commented on chef/cookbooks/monasca/recipes/agent.rb:
```diff
@@ -14,38 +14,111 @@
+cb = run_context.cookbook_collection[@cookbook_name]
 
-agent_settings = node[:monasca][:agent][:config]
+cb.manifest["templates"].each do |cbt|
```
I think that we are making a mistake here.

Basically hardcoding stuff you with to monitor (processes, http checks and so on) makes an assumption that you those applications, services etc will be always there. Which is basically not true.

I am afraid we cannot copy ansible-monasca-agent directly 1:1 here because there's no clear concept of host_vars + deployment is somehow dynamically mastered from UI (even if our own installer follows different principle right now). So I would drop all of those templates and rely purely on **full auto-detection** here and if we managed to find an app or whatever that is not supported in anyway we can always either:

* extend existing plugin of monasca-agent
* write new plugin

matrixik replied:
Chef will no longer check/manage folder `/etc/monasca/agent/conf.d/`

kornicameister reviewed (changes requested):
Part that I don't like is the one where you try to copy custom templates. That has little chance of working reliably.

matrixik committed:
Rename monasca-agent to monasca-metric-agent

And rename all variables to reflect this change.

matrixik committed:
Fix hound warnings

matrixik committed:
Merge remote-tracking branch 'fuji/monasca' into monasca

samirjorina commented on chef/cookbooks/monasca/attributes/default.rb:
```diff
@@ -17,11 +17,15 @@
   packages: [],
-  services: [monasca_log_agent_service_name]
+  services: [
+    monasca_agent_service_name,
```
the variable name is `monasca_metric_agent_service_name`

samirjorina commented on chef/cookbooks/monasca/recipes/metric_agent.rb:
```diff
@@ -0,0 +1,79 @@
+
+agent_dimensions = { service: "monitoring" }
+service = "monitoring"
+monasca_url = "#{monasca_protocol}://#{monasca_public_host}/v3"
```
What about port?

matrixik replied:
Done

samirjorina commented on chef/cookbooks/monasca/recipes/metric_agent.rb:
```diff
@@ -0,0 +1,79 @@
+service = "monitoring"
+monasca_url = "#{monasca_protocol}://#{monasca_public_host}/v3"
+
+template "/etc/sudoers.d/monasca-agent_sudoers" do
```
IHMO the `_sudoers` suffix is redundant, I'd remove it

samirjorina commented on crowbar_framework/app/helpers/barclamp/monasca_helper.rb:
```diff
@@ -0,0 +1,32 @@
+#
+# Copyright 2011-2013, Dell
```
Maybe it should be Copyright 2015 FUJITSU LIMITED?

matrixik committed:
Address comments

matrixik committed:
Merge remote-tracking branch 'fuji/monasca' into monasca

matrixik committed:
Fix copyright

matrixik committed:
Use same order in data_bag schema like in json

matrixik committed:
Restore user mapping in log_agent schema

matrixik committed:
Change monasca-master unique to false

matrixik committed:
Fix monasca-metric-agents names in json

matrixik committed:
Use keystone settings from data_bag for metric agent

dirkmueller commented on chef/cookbooks/monasca/recipes/metric_agent.rb:
```diff
@@ -0,0 +1,81 @@
+service = "monitoring"
+monasca_url = "#{monasca_protocol}://#{monasca_public_host}/v3"
+
+template "/etc/sudoers.d/monasca-agent" do
```
This belongs into packaging not in the barclamp

matrixik replied:
Well, I did not found any `sudo` in rpm-packaging however I found it in crowbar-openstack...:
https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/nova/recipes/monitor.rb#L50-L55
But we will move it if you want.

witekest replied:
I have added sudoers configuration [here](https://review.openstack.org/436063).

matrixik replied:
Code for sudoers removed from barclamp

artur-ba commented on chef/cookbooks/monasca/attributes/default.rb:
```diff
@@ -36,6 +40,12 @@
+# metric-agent default service settings
+default[:monasca][:metric_agent][:service_name] = monasca_metric_agent_service_name
+default[:monasca][:metric_agent][:user] = "monasca-agent"
+default[:monasca][:metric_agent][:group] = "monasca-agent"
```
Group should be monasca. As we discussed with other guys from community about rpm packaging.

matrixik committed:
Replace `owner` with `user` in execute command

matrixik committed:
Replace group `monasca-agent` with `monasca`

matrixik committed:
Remove adding monasca-agent to sudoers

Will be added from rpm.

matrixik commented:
Added SLES to supported dists to monasca-agent:
https://review.openstack.org/#/c/442381/

Otherwise barclamp is erroring with:
```
[2017-03-07T11:30:16+01:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: execute[monasca-setup detect services] (monasca::metric_agent line 56) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of /usr/local/bin/monasca-reconfigure ----
STDOUT:
STDERR: WARNING: SUSE Linux Enterprise Server is not a support Linux distribution
Traceback (most recent call last):
File "/usr/bin/monasca-setup", line 10, in
sys.exit(main())
File "/usr/lib/python2.7/site-packages/monasca_setup/main.py", line 50, in main
agent_service.enable()
File "/usr/lib/python2.7/site-packages/monasca_setup/service/linux.py", line 62, in enable
LinuxInit.enable(self)
File "/usr/lib/python2.7/site-packages/monasca_setup/service/linux.py", line 37, in enable
os.chown(self.log_dir, user.pw_uid, user.pw_gid)
OSError: [Errno 1] Operation not permitted: '/var/log/monasca/agent'
---- End output of /usr/local/bin/monasca-reconfigure ----
Ran /usr/local/bin/monasca-reconfigure returned 1 
```

matrixik committed:
Add port to monasca_api_url for monasca-agent

matrixik committed:
Use helper for getting api url

matrixik committed:
Fix bad default config name
